File: VB/GridControlDetailView/MyVGrid.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms
Imports CustomGrid

Namespace CustomGridControl
	Partial Public Class MyVGrid
		Inherits UserControl
		Implements IDetailView
		Public Sub New()
			InitializeComponent()
		End Sub

		Public Property DataSource() As Object Implements IDetailView.DataSource
			Get
				Return vGridControl1.DataSource
			End Get
			Set(ByVal value As Object)
				vGridControl1.DataSource = value
			End Set
		End Property

		Public ReadOnly Property InternalControl() As Control Implements IDetailView.InternalControl
			Get
				Return vGridControl1
			End Get
		End Property

		Private detailHeight_Renamed As Integer = 95
		Public ReadOnly Property DetailHeight() As Integer Implements IDetailView.DetailHeight
			Get
				Return detailHeight_Renamed
			End Get
		End Property
	End Class
End Namespace

File: CS/GridControlDetailView/MyVGrid.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using CustomGrid;

namespace CustomGridControl
{
    public partial class MyVGrid : UserControl, IDetailView
    {
        public MyVGrid()
        {
            InitializeComponent();
        }

        public object DataSource
        {
            get
            {
                return vGridControl1.DataSource;
            }
            set
            {
                vGridControl1.DataSource = value;
            }
        }

        public Control InternalControl
        {
            get
            {
                return vGridControl1;
            }
        }

        int detailHeight = 95;
        public int DetailHeight
        {
            get { return detailHeight; }
        }
    }
}
